### ISSUE TITLE
object-curly-newline is unusable

### ISSUE THREAD

**Issue by M1CK431** [eslint/eslint#13559]
**What rule do you want to change?**
object-curly-newline

**Does this change cause the rule to produce more or fewer warnings?**
more warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**

with this options object: `"object-curly-newline": ["warn", { multiline: true }]`:
```js
const obj = { a: "a value with a very long text which will exceed 80 characters and so should be break in multiple lines" }
```

**What does the rule currently do for this code?**
nothing

**What will the rule do after it's changed?**
based on this options object: `"object-curly-newline": ["warn", { multiline: true, maxLength: 80 }]`, break it in multiple lines
example:
```js
const obj = {
  a: "a value with a very long text which will exceed 80 characters and so should be break in multiple lines"
}
```
Also notice that, even if the property line length exceed the `maxLength` option **after** applying the multiline option, the rule should not emit a warning about that.

**Are you willing to submit a pull request to implement this change?**
no

Currently this rule as option to break on multiple lines depending on the number of properties in the object (`minProperties`) but there is **no option to break on multiple lines based on the line length** and so this rule is unusable (especially when used with other tools like Prettier for example).

Can you please add an option named `maxLength` (for example) which allow to define maximal line length for a one line object?

**Comment by anikethsaha** [comment]
Thanks for the issue. 

`object-curly-newline` is a stylistic rule and as per new policy , any new enhancement in stylistic rules are [frozen](https://eslint.org/blog/2020/05/changes-to-rules-policies)

You can always create a custom rule/plugin by extending this rule or create a new one.

**Comment by M1CK431** [comment]
Hi @anikethsaha and thanks for your quick reply.

I was not aware about the new policy.

I have read the blog post you linked and I don't find any information about the end of the freeze or a plan about the futur of existing stylistic rules (for example something like "we will extract them in a plugin which will be manage by the community").

Also, the blog post speak about "numerous tutorials have made it easier than ever for developers to write their own rules" but there is no reference or link to any of them.

Could you advise me for a good tutorial which will help me to write my own enhanced version of object-curly-newline rule?

**Comment by anikethsaha** [comment]
I am not sure about the tutorials link. 
But I hope these references can be helpful 

- use this rule as base rule and extend the functionality [ref this](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/brace-style.ts)
- [eslint-rule-composer](https://github.com/not-an-aardvark/eslint-rule-composer)
- [use this playground for better experience](https://astexplorer.net/)
- [may be a useful video](https://www.youtube.com/watch?v=VicU_XwriWw)

**Comment by M1CK431** [comment]
Ok thanks.

After checking all of them, I understand that it simply too complicated for me.

To make this change I have to:
- learn what is AST and how it works
- create a dedicated repo/package (my plugin)
- add my only rule
- add it inside all my project...

All of this just for a missing option... :exploding_head:

IMHO, the lake of this option could be consider as a "bug" at conception time because I guess that every projets in the world have a max length rule (configured in eslint, prettier or whatever) and so providing this kind of styling rule without taking care of that make it just unusable in most case.

Finally, [like many others](https://github.com/search?p=1&q=object-curly-newline&type=Commits) and even if it make me really sad, I'm forced to disable this rule in my projet and live with inconsistent code.

Thanks anyway @anikethsaha for your quick reply and help. I really hope that someone with the required knowledge will finally update `object-curly-newline` rule in a way that everyone will be able to configure it accordingly with the max length rule of the projet.

**Comment by anikethsaha** [comment]
Thanks for understanding. 

You can use `no-restricted-syntax` to report that but I am not sure about the autofix
```js
/* eslint no-restricted-syntax: ["error", {selector: "ObjectExpression > Property[value.value=/^.{80,}$/]", message: "more than 80 chars, add a new line"}] */

```

[check the demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7c2VsZWN0b3I6IFwiT2JqZWN0RXhwcmVzc2lvbiA+IFByb3BlcnR5W3ZhbHVlLnZhbHVlPS9eLns4MCx9JC9dXCIsIG1lc3NhZ2U6IFwibW9yZSB0aGFuIDgwIGNoYXJzLCBhZGQgYSBuZXcgbGluZVwifV0gKi9cblxuY29uc3Qgb2JqID0geyBiOidhJyxhOiBcImEgdmFsdWUgd2l0aCBhIHZlcnkgbG9uZyB0ZXh0IHdoaWNoIHdpbGwgZXhjZWVkIDgwIGNoYXJhY3RlcnMgYW5kIHNvIHNob3VsZCBiZSBicmVhayBpbiBtdWx0aXBsZSBsaW5lc1wiIH0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImluZGVudCI6Mn0sImVudiI6e319fQ==)

> IMHO, the lake of this option could be consider as a "bug" at conception time because I guess that every projets in the world have a max length rule (configured in eslint, prettier or whatever) and so providing this kind of styling rule without taking care of that make it just unusable in most case.

the bug can be considered only if the rule is not working properly or creating false results or causing errors in builds.
This is an enhancement request.

**Comment by M1CK431** [comment]
In fact i'm thinking about another way to achieve the same goal: create another rule ID in addition to `0` ("off"), `1` ("warning") and `2` ("error"). It could be named `quiet`, `silence`, `discret` or even `ghost` and the behavior is: the rule is enabled (used to auto-fix) but doesn't display any message in console.

There is some huge advantage with this approach:
- could be apply on any rules
- isn't concern by "styling rule freeze policy"

Wdyt about this idea @anikethsaha ?

**Comment by anikethsaha** [comment]
Personally , I have never encountered a situation where I didn't want to make use of the logs in the console. So I dont have strong opinion about this. Not all rules are auto-fixable and this would need a change in core and cli . 

If you want, you cant create an issue/RFC with the detailed approach and problem explanation and others can give their inputs in that as well.
